File: warden-api/warden-api/Application/Module/Ci/Model/CiScanRequest.cs
using System.ComponentModel.DataAnnotations;
using Warden.Application.Validators;
using Warden.Core.Enum;

namespace Warden.Application.Module.Ci.Model;

public record CiScanRequest
{
    [Required] public required SourceType Source { get; set; }
    [Required] public required string RepoId { get; set; }

    [Required] 
    [HttpUrl]
    public required string RepoUrl { get; set; }

    public string RepoName { get; set; } = string.Empty;

    [Required] public required CommitType GitAction { get; set; }

    [Required] public required string ScanTitle { get; set; }

    public required string CommitBranch { get; set; }

    [Required] public required string CommitHash { get; set; }

    public string? TargetBranch { get; set; }
    public string? MergeRequestId { get; set; }

    [Required] public required string Scanner { get; set; }

    [Required] public required ScannerType Type { get; set; }

    public string JobUrl { get; set; } = string.Empty;

    public required bool IsDefault { get; set; }
    public string? ContainerImage { get; set; }
}

File: scanners/quarkus/warden-client/src/main/java/com/techanv/warden/client/model/CiScanRequest.java
package com.techanv.warden.client.model;

import com.fasterxml.jackson.annotation.JsonInclude;

/** Request body for {@code POST /api/ci/scan}. Mirrors the .NET CiScanRequest. */
@JsonInclude(JsonInclude.Include.NON_NULL)
public class CiScanRequest {
    public String source = "Local";
    public String repoId;
    public String repoUrl;
    public String repoName;
    public String gitAction = "CommitBranch";
    public String scanTitle;
    public String commitBranch;
    public String commitHash;
    public String targetBranch;
    public String mergeRequestId;
    public String scanner;
    public String type;
    public String jobUrl;
    public boolean isDefault = true;
    public String containerImage;
}
